pkg/cloud/scope/machine.go

```diff
@@ -275,6 +276,10 @@ func (m *MachineScope) GetRawBootstrapDataWithFormat() ([]byte, string, error) {
 	secret := &corev1.Secret{}
 	key := types.NamespacedName{Namespace: m.Namespace(), Name: *m.Machine.Spec.Bootstrap.DataSecretName}
 	if err := m.client.Get(context.TODO(), key, secret); err != nil {
+		if apierrors.IsNotFound(err) {
```

Comment:
@enxebre is it possible to add a unit test for the same?

Reply:
maybe also add a debug log?

Reply:
@enxebre can these comments be taken care of?